packages/terra-form-input/src/terra-dev-site/doc/example/controlled/InputProgrammaticSet.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useRef } from 'react';
+import classNames from 'classnames/bind';
+import Input from 'terra-form-input';
+import styles from './InputProgrammaticSet.module.scss';
+
+const cx = classNames.bind(styles);
+
+const InputProgrammaticSet = () => {
+  const [text, setText] = useState('');
+  const inputRef = useRef(null);
+
+  const setTextClick = () => {
+    if (inputRef.current) {
+      inputRef.current.value = 'IpsumLorem';
+      const customEvent = new Event('input', {
+        bubbles: true,
+        cancelable: false,
+      });
+      inputRef.current.dispatchEvent(customEvent);
```

Comment:
why do we need to dispatch a event here..? Does changing the value programmatically doesn't triggers `onInput` for Input element..?

Reply:
Yup, changing value programmatically doesn't trigger any event. 
And Nuance dictation is programmed in such a way that it creates an `input` event to be fired after they programmatically set the value with dictation.